### PR TITLE
PR: Fix communication between service SG and ALB SG when alb is created, set egress by default

### PR DESCRIPTION
**Pull request by patrickmukumbu**
# Description
This PR was created to fix the communication between service security group and load balancer security group, when the lb SG is created. If no LB is created, a SG for internal communication within the service is created.

## Features/Fixes/Patches/Changes list:

Please check the [CHANGELOG.md](https://github.com/boldlink/terraform-aws-ecs-service/blob/fix/security-group-connections/CHANGELOG.md#151---2023-06-06)

### PR Owners Checklist:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you checked there aren't open [Pull Requests](../../../pulls) of upstream or parallel code that will cause conflicts?
* [x] Have you updated the `CHANGELOG.md`?
* [x] Have you updated the `README.md`(s)?
* [x] OWNER: Does your submission pass?
    * [x] Pre-commit (attach logs/print screen to this PR)
    * [x] Checkov/terrascan (attach logs/print screen to this PR)
    * [x] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] First PR? Have you deleted `## How to use this template...` from the root `.README.md`?

### PR Reviewers Checklists?
* [ ] Are all your comments/changes resolved?
* [ ] Are you happy with the OWNER checklists and additional information provided?
* [ ] Does your review tests pass?
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -9,12 +9,14 @@ module "access_logs_bucket" {
 
 module "ecs_service_lb" {
   #checkov:skip=CKV_AWS_150: "Ensure that Load Balancer has deletion protection enabled"
+  #checkov:skip=CKV_AWS_290: "Ensure IAM policies does not allow write access without constraints"
```
Comment (hugoalmeida264) [inline examples/complete/main.tf:12]: These exceptions are missing from the changelog

**Review by hugoalmeida264 — CHANGES_REQUESTED** [review]
- Missing changelog checkov exceptions
- trigger property missing to enable the force new deployment, this should also be documented on README that the module supports force deployment and what are the scenarios for it, ex when you are using the same image TAG instead of semantic versioning.

File: examples/complete/main.tf (hunk 1 above)
Comment (patrickmukumbu) [reply examples/complete/main.tf:12]: Addressed @hugoalmeida264

File: examples/fargate/main.tf [hunk 2]
```diff
@@ -1,4 +1,6 @@
 module "ecs_service" {
+  #checkov:skip=CKV_AWS_290: "Ensure IAM policies does not allow write access without constraints"
+  #checkov:skip=CKV_AWS_355: "Ensure no IAM policies documents allow "*" as a statement's resource for restrictable actions"
```
Comment (hugoalmeida264) [inline examples/fargate/main.tf:3]: 355 missing from changelog

File: main.tf [hunk 3]
```diff
@@ -34,7 +35,10 @@ resource "aws_ecs_service" "service" {
     }
   }
 
-  tags = var.tags
+  triggers = {
```
Comment (hugoalmeida264) [inline main.tf:38]: This is hardcoded when it should be a dynamic field, you can then use the `timestamp()` instead on the example.

**Review by hugoalmeida264 — CHANGES_REQUESTED** [review]

**Pull request merged**